great_expectations/experimental/datasources/sources.py

```diff
@@ -213,6 +214,9 @@ def _add_asset_factory(
             )
             setattr(ds_type, add_asset_factory_method_name, _add_asset_factory)
 
+            # add the public api decorator
+            public_api(getattr(ds_type, add_asset_factory_method_name))
+
```

Comment:
This won't work, because our method of scanning for the `public_api` tag works by parsing the sourcecode, not executing the code.